### PR TITLE
Trap signals rather than killing processes

### PR DESCRIPTION
**Pull request by joelvh**
We were debugging rogue processes and didn't realize the package had a fix for the issue already. However, we believe the more "native" solution here is to leverage `Watch::shouldContinue` with Laravel's built-in `trap` for signal trapping.

This simplifies the code and doesn't run another shell command.

BTW we have been using [spatie/laravel-signal-aware-command](https://github.com/spatie/laravel-signal-aware-command) and found that it did not work as expected with signal trapping, and found the native solution works more reliably, as seen in this solution.

### CODE REVIEW

File: src/Commands/WatchHorizonCommand.php [hunk 1]
```diff
@@ -10,14 +10,20 @@ class WatchHorizonCommand extends Command
 {
     protected Process $horizonProcess;
 
-    protected $signature = 'horizon:watch {--without-tty : Disable output to TTY}}';
+    protected $signature = 'horizon:watch {--without-tty : Disable output to TTY}';
```
Comment (joelvh) [inline src/Commands/WatchHorizonCommand.php:13]: Also fixes a typo here.

**Comment by freekmurze** [comment]
Very nice, thank you!

**Comment by sdapkus** [comment]
For me this doesn't work. After making some changes in a project and killing horizon I get this log:
```bash
$ sail artisan horizon:watch

   INFO  Starting Horizon and will restart it when any files change...  

Horizon started successfully.
   INFO  Change detected! Restarting horizon...  

Horizon started successfully.
   INFO  Change detected! Restarting horizon...  

Horizon started successfully.
   INFO  Change detected! Restarting horizon...  

Horizon started successfully.
^CShutting down...
Shutting down...
Shutting down...
Shutting down...
```

Also in a Horizon dashboard I can see that after each change new worker is started.

image


I have double check 1.0.4 version - all worker processes between reloads seems to be killed normally.
My environment:
- Mac M1
- Laravel Sail 1.25.0
- Laravel 10.23.1
- Horizon 5.21.0

**Comment by freekmurze** [comment]
@joelvh could you have a look?

**Comment by joelvh** [comment]
Hi @sdapkus @freekmurze,

It appears that this is a bug that was already present. This PR addressed signal trapping, but didn't account for the signal forwarding piece that the previous process killing was addressing.

At some point, we generalized this watcher command to watch any process we want, and we fixed it in our version of the command. FWIW, I wasn't aware that it was something that was a pre-existing bug here as well.

The fix in #22 should do the trick. @sdapkus if you could please verify. Thanks!

**Comment by binaryfire** [comment]
We're also experiencing this when running Horizon inside a container: https://github.com/spatie/laravel-horizon-watcher/issues. Has anyone found a workaround? We've got a remote team and our app has a lot of services, so running Horizon outside of our Docker development environment isn't an option unfortunately.

**Pull request merged**